types/src/protocol_state.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use mina_crypto::hash::BaseHash;
+use serde::{Deserialize, Serialize};
+
+// TODO: confirm below types
+pub type Length = u64;
+pub type VrfOutput = [u8; 32];
+pub type Amount = u64;
+
+/// GlobalSlot contains (slot_number, slots_per_epoch)
+pub type GlobalSlot = (u64, u64);
```

Comment:
Quite a few of these types are already defined in the base crate: https://github.com/ChainSafe/mina-rs/blob/main/base/src/global_slot.rs We can just import and use them.

Reply:
wow I totally missed this, will update